BLSQ/iaso
# POLIO-1236 add automatic check on the quantity entered in campaigns per country vs quantity authorized

Check if the quantity of requested doses in Risk Assessment campaign is inferior or equal to the authorized doses in vaccine authorization for the country. When creating a campaign, if there is an authorization the check operate. If not, a mail is sent to warn the NOPV2 Team that a campaign without authorization has been created. 

When updating a campaign the check also operate. 

There was also a bug if no NOPV2 team was created, when sending the mail it was making the campaign creation crash. ( Wont happen in prod but we never know... )

Related JIRA tickets : POLIO-1236

## Self proofreading checklist


## Changes

## How to test

You need to have a team named " nOPV2 vaccine authorization alerts " in order to carry the test.
First scenario : 
Create a campaign without vaccine authorization approved, add a round and in risk assessement add a quantity for this round. You should be able to create the campaign. 
A mail should be sent to warn that a campaign with vaccine haut has been created.
Then create  a vaccine authorization, add X quantity. Go back to the campaign, add a round and test the quantity. If the quantity is superior to the vacc auth, an error should be triggered. 

I also suggest you to test all possibilities of the campaign as it touch directly the campaign API code. 

Second Scenario: 

Create an approved vaccine auth first with X quantity. Create a campaign, add a round and add requested doses superior to the vacc auth qty. An error should be triggered. 


https://github.com/BLSQ/iaso/assets/45785235/9057f9c0-337b-4750-814b-d4d61112f822
